### ISSUE TITLE
permission denied (public key) error trying to run git submodule init

### ISSUE THREAD

**Issue by obuckley-kenzan** [kenzanlabs/million-song-library#347]
After cloning the repo and trying to run `git submodule init` a permission denied error is seen trying to clone all the sub projects
```
1                    os-x-10:~ markanthony$ git clone https://github.com/kenzanlabs/million-song-library
2                    Cloning into 'million-song-library'...
3                    remote: Counting objects: 10018, done.
4                    remote: Compressing objects: 100% (5/5), done.
5                    remote: Total 10018 (delta 1), reused 0 (delta 0), pack-reused 10013
6                    Receiving objects: 100% (10018/10018), 103.65 MiB | 9.82 MiB/s, done.
7                    Resolving deltas: 100% (5643/5643), done.
8                    os-x-10:~ markanthony$ cd million-song-library/
9                    os-x-10:million-song-library markanthony$ git submodule init && git submodule update
10                Submodule 'msl-pages' (git@github.com:kenzanlabs/msl-pages.git) registered for path 'msl-pages'
11                Submodule 'server/msl-account-data-client' (git@github.com:kenzanlabs/msl-account-data-client.git) registered for path 'server/msl-account-data-client'
12                Submodule 'server/msl-account-data-client-config' (git@github.com:kenzanlabs/msl-account-data-client-config.git) registered for path 'server/msl-account-data-client-config'
13                Submodule 'server/msl-account-edge' (git@github.com:kenzanlabs/msl-account-edge.git) registered for path 'server/msl-account-edge'
14                Submodule 'server/msl-account-edge-config' (git@github.com:kenzanlabs/msl-account-edge-config.git) registered for path 'server/msl-account-edge-config'
15                Submodule 'server/msl-catalog-data-client' (git@github.com:kenzanlabs/msl-catalog-data-client.git) registered for path 'server/msl-catalog-data-client'
16                Submodule 'server/msl-catalog-data-client-config' (git@github.com:kenzanlabs/msl-catalog-data-client-config.git) registered for path 'server/msl-catalog-data-client-config'
17                Submodule 'server/msl-catalog-edge' (git@github.com:kenzanlabs/msl-catalog-edge.git) registered for path 'server/msl-catalog-edge'
18                Submodule 'server/msl-catalog-edge-config' (git@github.com:kenzanlabs/msl-catalog-edge-config.git) registered for path 'server/msl-catalog-edge-config'
19                Submodule 'server/msl-login-edge' (git@github.com:kenzanlabs/msl-login-edge.git) registered for path 'server/msl-login-edge'
20                Submodule 'server/msl-login-edge-config' (git@github.com:kenzanlabs/msl-login-edge-config.git) registered for path 'server/msl-login-edge-config'
21                Submodule 'server/msl-ratings-data-client' (git@github.com:kenzanlabs/msl-ratings-data-client.git) registered for path 'server/msl-ratings-data-client'
22                Submodule 'server/msl-ratings-data-client-config' (git@github.com:kenzanlabs/msl-ratings-data-client-config.git) registered for path 'server/msl-ratings-data-client-config'
23                Submodule 'server/msl-ratings-edge' (git@github.com:kenzanlabs/msl-ratings-edge.git) registered for path 'server/msl-ratings-edge'
24                Submodule 'server/msl-ratings-edge-config' (git@github.com:kenzanlabs/msl-ratings-edge-config.git) registered for path 'server/msl-ratings-edge-config'
25                Submodule 'server/msl-server-common' (git@github.com:kenzanmedia/msl-server-common.git) registered for path 'server/msl-server-common'
26                Submodule 'server/msl-zuul' (git@github.com:kenzanlabs/msl-zuul.git) registered for path 'server/msl-zuul'
27                Submodule 'server/styleguide' (https://github.com/kenzanlabs/styleguide.git) registered for path 'server/styleguide'
28                Cloning into '/Users/markanthony/million-song-library/msl-pages'...
29                The authenticity of host 'github.com (192.30.253.113)' can't be established.
30                RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
31                Are you sure you want to continue connecting (yes/no)? yes
32                Warning: Permanently added 'github.com,192.30.253.113' (RSA) to the list of known hosts.
33                Permission denied (publickey).
34                fatal: Could not read from remote repository.
35                 
36                Please make sure you have the correct access rights
37                and the repository exists.
38                fatal: clone of 'git@github.com:kenzanlabs/msl-pages.git' into submodule path '/Users/markanthony/million-song-library/msl-pages' failed
39                Failed to clone 'msl-pages'. Retry scheduled
40                Cloning into '/Users/markanthony/million-song-library/server/msl-account-data-client'...
41                Warning: Permanently added the RSA host key for IP address '192.30.253.112' to the list of known hosts.
42                Permission denied (publickey).
43                fatal: Could not read from remote repository.
44                 
45                Please make sure you have the correct access rights
46                and the repository exists.
47                fatal: clone of 'git@github.com:kenzanlabs/msl-account-data-client.git' into submodule path '/Users/markanthony/million-song-library/server/msl-account-data-client' failed
48                Failed to clone 'server/msl-account-data-client'. Retry scheduled
49                Cloning into '/Users/markanthony/million-song-library/server/msl-account-data-client-config'...
50                Permission denied (publickey).
51                fatal: Could not read from remote repository.
52                 
53                Please make sure you have the correct access rights
54                and the repository exists.
55                fatal: clone of 'git@github.com:kenzanlabs/msl-account-data-client-config.git' into submodule path '/Users/markanthony/million-song-library/server/msl-account-data-client-config' failed
56                Failed to clone 'server/msl-account-data-client-config'. Retry scheduled
57                Cloning into '/Users/markanthony/million-song-library/server/msl-account-edge'...
58                Permission denied (publickey).
59                fatal: Could not read from remote repository.
60                 
61                Please make sure you have the correct access rights
62                and the repository exists.
63                fatal: clone of 'git@github.com:kenzanlabs/msl-account-edge.git' into submodule path '/Users/markanthony/million-song-library/server/msl-account-edge' failed
64                Failed to clone 'server/msl-account-edge'. Retry scheduled
65                Cloning into '/Users/markanthony/million-song-library/server/msl-account-edge-config'...
66                Permission denied (publickey).
67                fatal: Could not read from remote repository.
68                 
69                Please make sure you have the correct access rights
70                and the repository exists.
71                fatal: clone of 'git@github.com:kenzanlabs/msl-account-edge-config.git' into submodule path '/Users/markanthony/million-song-library/server/msl-account-edge-config' failed
72                Failed to clone 'server/msl-account-edge-config'. Retry scheduled
73                Cloning into '/Users/markanthony/million-song-library/server/msl-catalog-data-client'...
74                Permission denied (publickey).
75                fatal: Could not read from remote repository.
76                 
77                Please make sure you have the correct access rights
78                and the repository exists.
79                fatal: clone of 'git@github.com:kenzanlabs/msl-catalog-data-client.git' into submodule path '/Users/markanthony/million-song-library/server/msl-catalog-data-client' failed
80                Failed to clone 'server/msl-catalog-data-client'. Retry scheduled
81                Cloning into '/Users/markanthony/million-song-library/server/msl-catalog-data-client-config'...
82                Permission denied (publickey).
83                fatal: Could not read from remote repository.
84                 
85                Please make sure you have the correct access rights
86                and the repository exists.
87                fatal: clone of 'git@github.com:kenzanlabs/msl-catalog-data-client-config.git' into submodule path '/Users/markanthony/million-song-library/server/msl-catalog-data-client-config' failed
88                Failed to clone 'server/msl-catalog-data-client-config'. Retry scheduled
89                Cloning into '/Users/markanthony/million-song-library/server/msl-catalog-edge'...
90                Permission denied (publickey).
91                fatal: Could not read from remote repository.
92                 
93                Please make sure you have the correct access rights
94                and the repository exists.
95                fatal: clone of 'git@github.com:kenzanlabs/msl-catalog-edge.git' into submodule path '/Users/markanthony/million-song-library/server/msl-catalog-edge' failed
96                Failed to clone 'server/msl-catalog-edge'. Retry scheduled
97                Cloning into '/Users/markanthony/million-song-library/server/msl-catalog-edge-config'...
98                Permission denied (publickey).
99                fatal: Could not read from remote repository.
100             
101            Please make sure you have the correct access rights
102            and the repository exists.
103            fatal: clone of 'git@github.com:kenzanlabs/msl-catalog-edge-config.git' into submodule path '/Users/markanthony/million-song-library/server/msl-catalog-edge-config' failed
104            Failed to clone 'server/msl-catalog-edge-config'. Retry scheduled
105            Cloning into '/Users/markanthony/million-song-library/server/msl-login-edge'...
106            Permission denied (publickey).
107            fatal: Could not read from remote repository.
108             
109            Please make sure you have the correct access rights
110            and the repository exists.
111            fatal: clone of 'git@github.com:kenzanlabs/msl-login-edge.git' into submodule path '/Users/markanthony/million-song-library/server/msl-login-edge' failed
112            Failed to clone 'server/msl-login-edge'. Retry scheduled
113            Cloning into '/Users/markanthony/million-song-library/server/msl-login-edge-config'...
114            Permission denied (publickey).
115            fatal: Could not read from remote repository.
116             
117            Please make sure you have the correct access rights
118            and the repository exists.
119            fatal: clone of 'git@github.com:kenzanlabs/msl-login-edge-config.git' into submodule path '/Users/markanthony/million-song-library/server/msl-login-edge-config' failed
120            Failed to clone 'server/msl-login-edge-config'. Retry scheduled
121            Cloning into '/Users/markanthony/million-song-library/server/msl-ratings-data-client'...
122            Permission denied (publickey).
123            fatal: Could not read from remote repository.
124             
125            Please make sure you have the correct access rights
126            and the repository exists.
127            fatal: clone of 'git@github.com:kenzanlabs/msl-ratings-data-client.git' into submodule path '/Users/markanthony/million-song-library/server/msl-ratings-data-client' failed
128            Failed to clone 'server/msl-ratings-data-client'. Retry scheduled
129            Cloning into '/Users/markanthony/million-song-library/server/msl-ratings-data-client-config'...
130            Permission denied (publickey).
131            fatal: Could not read from remote repository.
132             
133            Please make sure you have the correct access rights
134            and the repository exists.
135            fatal: clone of 'git@github.com:kenzanlabs/msl-ratings-data-client-config.git' into submodule path '/Users/markanthony/million-song-library/server/msl-ratings-data-client-config' failed
136            Failed to clone 'server/msl-ratings-data-client-config'. Retry scheduled
137            Cloning into '/Users/markanthony/million-song-library/server/msl-ratings-edge'...
138            Permission denied (publickey).
139            fatal: Could not read from remote repository.
140             
141            Please make sure you have the correct access rights
142            and the repository exists.
143            fatal: clone of 'git@github.com:kenzanlabs/msl-ratings-edge.git' into submodule path '/Users/markanthony/million-song-library/server/msl-ratings-edge' failed
144            Failed to clone 'server/msl-ratings-edge'. Retry scheduled
145            Cloning into '/Users/markanthony/million-song-library/server/msl-ratings-edge-config'...
146            Permission denied (publickey).
147            fatal: Could not read from remote repository.
148             
149            Please make sure you have the correct access rights
150            and the repository exists.
151            fatal: clone of 'git@github.com:kenzanlabs/msl-ratings-edge-config.git' into submodule path '/Users/markanthony/million-song-library/server/msl-ratings-edge-config' failed
152            Failed to clone 'server/msl-ratings-edge-config'. Retry scheduled
153            Cloning into '/Users/markanthony/million-song-library/server/msl-server-common'...
154            Permission denied (publickey).
155            fatal: Could not read from remote repository.
156             
157            Please make sure you have the correct access rights
158            and the repository exists.
159            fatal: clone of 'git@github.com:kenzanmedia/msl-server-common.git' into submodule path '/Users/markanthony/million-song-library/server/msl-server-common' failed
160            Failed to clone 'server/msl-server-common'. Retry scheduled
161            Cloning into '/Users/markanthony/million-song-library/server/msl-zuul'...
162            Permission denied (publickey).
163            fatal: Could not read from remote repository.
164             
165            Please make sure you have the correct access rights
166            and the repository exists.
167            fatal: clone of 'git@github.com:kenzanlabs/msl-zuul.git' into submodule path '/Users/markanthony/million-song-library/server/msl-zuul' failed
168            Failed to clone 'server/msl-zuul'. Retry scheduled
169            Cloning into '/Users/markanthony/million-song-library/server/styleguide'...
170            Cloning into '/Users/markanthony/million-song-library/msl-pages'...
171            Permission denied (publickey).
172            fatal: Could not read from remote repository.
173             
174            Please make sure you have the correct access rights
175            and the repository exists.
176            fatal: clone of 'git@github.com:kenzanlabs/msl-pages.git' into submodule path '/Users/markanthony/million-song-library/msl-pages' failed.
177            Failed to clone 'msl-pages' a second time, aborting
```

cc: @bschwanitz

**Comment by obuckley-kenzan** [comment]
It appears that all the sub modules have their URL set to SSH. 
https://github.com/kenzanlabs/million-song-library/blob/develop/.gitmodules

It should be changed to HTTP.  So for example, for MSL Server Common
https://github.com/kenzanlabs/million-song-library/blob/develop/.gitmodules#L3

the format would need to be
`https://github.com/kenzanlabs/msl-server-common.git`

Let me know if you have any questions.

**Comment by obuckley-kenzan** [comment]
resolved by #348
